crypto/zkp_bip340.c

```diff
@@ -157,3 +164,148 @@ int zkp_bip340_verify_digest(const uint8_t *public_key_bytes,
 
   return result;
 }
+
+// BIP340 Schnorr public key tweak
+// internal_public_key has 32 bytes
+// root_hash has 32 bytes or is empty (NULL)
+// output_public_key has 32 bytes
+// returns 0 on success
+int zkp_bip340_tweak_public_key(const uint8_t *internal_public_key,
+                                const uint8_t *root_hash,
+                                uint8_t *output_public_key) {
+  int result = 0;
+
+  uint8_t tweak[SHA256_DIGEST_LENGTH] = {0};
+  if (result == 0) {
+    SHA256_CTX ctx = {0};
+    sha256_Init_ex(&ctx, sha256_initial_taptweak_state, 512);
```

Comment:
I think `bitcount` should be set to zero (instead of 512), because `bitcount` indicates how much unprocessed data is in the context buffer. See also https://github.com/trezor/trezor-firmware/blob/a01516389a5a98f0d63205fdd40f8907e2054f87/crypto/sha2.c#L677.

Reply:
> `bitcount` indicates how much unprocessed data is in the context buffer

Not true. `bitcount` is the number of bits of data that has been fed into `sha256_Update()`.

See also https://github.com/trezor/trezor-firmware/pull/1891#discussion_r742692147 and
https://satoshilabs.slack.com/archives/CL1D61PQF/p1633361230093600?thread_ts=1632922521.089300&cid=CL1D61PQF.